src/config.ts

```diff
@@ -75,25 +76,44 @@ const defaultConfig: ConfigInterface = {
 }
 
 // Focus revalidate
-let eventsBinded = false
-if (typeof window !== 'undefined' && window.addEventListener && !eventsBinded) {
-  const revalidate = () => {
-    if (!isDocumentVisible() || !isOnline()) return
+if (typeof window !== 'undefined' && window.addEventListener) {
+  // only bind the events once
+  let focusEventsBinded = false
+  let onlineEventsBinded = false
+  if (!focusEventsBinded) {
+    const revalidate = () => {
+      if (!isDocumentVisible() || !isOnline()) return
 
-    for (let key in FOCUS_REVALIDATORS) {
-      if (FOCUS_REVALIDATORS[key][0]) FOCUS_REVALIDATORS[key][0]()
+      for (const key in FOCUS_REVALIDATORS) {
+        if (FOCUS_REVALIDATORS[key][0]) FOCUS_REVALIDATORS[key][0]()
+      }
     }
+    window.addEventListener('visibilitychange', revalidate, false)
+    window.addEventListener('focus', revalidate, false)
+
+    focusEventsBinded = true
+  }
+
+  // set up reconnecting when the browser regains network connection
+  if (defaultConfig.revalidateOnReconnect && !onlineEventsBinded) {
+    const revalidate = () => {
+      if (!isOnline()) return
```

Comment:
I don't know if we need `isDocumentVisible` condition here, so I left a comment for checking.